test/create-placeholders-for-phi-operands.ll

```diff
@@ -31,7 +31,7 @@ BB.0:
   br i1 %or.cond, label %BB.2, label %BB.3
 
 BB.1:                                             ; preds = %BB.12, %BB.7
-  %savedstack.sink = phi i8* [ %savedstack, %BB.12 ], [ %savedstack.us, %BB.7 ]
+  %savedstack.sink = phi i8* [ %savedstack, %BB.12.loopexit ], [ %savedstack.us, %BB.11.loopexit ]
```

Comment:
Could you please explain this change? TBH I wouldn't expect source IR for test to be changed - only IR after the translation.
This comment applies to every other IR change in the tests.

Reply:
My assumption from the name of this test is that the intention is to make sure there aren't problems with round-tripping phi nodes that appear before their non-dominating operands in the function. The problem that I ran into is that the translator adds these two loopexit blocks ahead of `BB.1` and dominator ordering pulls all of the loopexit blocks' dominators ahead of them (including the ones that define `savedstack` and `savedstack.us`). The result is that this phi no longer appears before the non-dominating operands it references, which seems to defeat the purpose of this test. My fix here is to create these loopexit blocks manually before the translator can and to place them later in the function so that they don't pull the phi operands ahead of the phi.